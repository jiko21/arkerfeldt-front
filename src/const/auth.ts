export const authCookie = 'auth-cookie';

export enum Provider {
  EMAIL,
  GITHUB,
}
