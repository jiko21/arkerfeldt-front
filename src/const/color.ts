export const Color = {
  BLACK: '#2d2d2d',
  LIGHT: '#878686',
  ENABLE: '#007AAE',
  ENABLE_HOVER: '#005b82',
  DISABLE: '#F0F0F0',
  DISABLE_HOVER: '#d1d1d1',
  LIGHT_GRAY: '#C4C4C4',
  LIGHT_TEXT: '#6A6A6A',
  WHITE: '#FFFFFF',
  WHITE_LIGHT: '#FCFCFC',
  WHITE_LIGHT_HOVER: '#ECECEC',
} as const;

export type Color = typeof Color[keyof typeof Color];
