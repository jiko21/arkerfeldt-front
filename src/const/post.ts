export const PublishStatus = {
  PUBLISHED: 'PUBLISHED',
  UNPUBLISHED: 'UNPUBLISHED',
} as const;

export type PublishStatus = typeof PublishStatus[keyof typeof PublishStatus];

export type Post = {
  id: number;
  title: string;
  content: string;
  createdAt: Date;
  updatedAt: Date;
  isDelete: boolean;
  status: PublishStatus;
  authorId: string;
  author: {
    displayName: string;
  };
};

export type PostInputType = {
  title: string;
  content: string;
  status: PublishStatus;
};
