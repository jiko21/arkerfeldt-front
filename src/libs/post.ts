import { PublishStatus } from '@/const/post';

export const convertStatusToString = (status: PublishStatus) => {
  switch (status) {
    case PublishStatus.PUBLISHED:
      return '公開中';
    case PublishStatus.UNPUBLISHED:
      return '未公開';
  }
};

export const convertStatusToText = (status: PublishStatus) => {
  switch (status) {
    case PublishStatus.PUBLISHED:
      return '公開する';
    case PublishStatus.UNPUBLISHED:
      return '未公開のまま保存';
  }
};
