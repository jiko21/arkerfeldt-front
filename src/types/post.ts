import { PublishStatus } from '@/const/post';

export type Post = {
  id: number;
  title: string;
  content: string;
  createdAt: Date;
  updatedAt: Date;
  isDelete: boolean;
  status: PublishStatus;
  authorId: string;
  author: {
    displayName: string;
  };
};

export type PostInputType = {
  title: string;
  content: string;
  status: PublishStatus;
};
