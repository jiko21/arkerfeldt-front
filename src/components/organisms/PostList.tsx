/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { Post } from '@/types/post';
import PostItem from '@/components/molecules/PostItem';
import * as React from 'react';

type Props = {
  posts: Post[];
};

const ulStyle = css`
  list-style: none;
  padding: 0;
  li {
    margin-bottom: 6px;
  }
`;

const PostList: React.FC<Props> = ({ posts }) => {
  return (
    <ul css={ulStyle}>
      {posts.map((post) => (
        <li key={post.id}>
          <PostItem
            post={post}
            checked={false}
            onChecked={(value: boolean) => {
              console.log(value);
            }}
            link={''}
          />
        </li>
      ))}
    </ul>
  );
};

export default PostList;
