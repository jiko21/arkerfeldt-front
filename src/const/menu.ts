import { MenuItem } from '@/types/MenuItem';

export const menuItems: MenuItem[] = [
  {
    title: 'Post',
    url: '/post',
    isSelected: true,
  },
  {
    title: 'Layout',
    url: '/layout',
    isSelected: false,
  },
  {
    title: 'User',
    url: '/user',
    isSelected: false,
  },
];
