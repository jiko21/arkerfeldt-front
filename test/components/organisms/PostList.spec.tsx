import PostList from '@/components/organisms/PostList';
import { PublishStatus } from '@/const/post';
import { render } from '@testing-library/react';

describe('PostList.spec.tsx', () => {
  it('should correctly render', () => {
    const props = {
      posts: [
        {
          id: 1,
          title: '〇〇は結婚していた？彼女の名前は××？',
          content: '',
          createdAt: new Date('2020/12/11 10:00'),
          updatedAt: new Date('2020/12/11 10:00'),
          isDelete: false,
          status: PublishStatus.PUBLISHED,
          authorId: '123',
          author: {
            displayName: 'jiko21',
          },
        },
        {
          id: 2,
          title: '〇〇は結婚していた？彼女の名前は××？',
          content: '',
          createdAt: new Date('2020/12/11 10:00'),
          updatedAt: new Date('2020/12/11 10:00'),
          isDelete: false,
          status: PublishStatus.UNPUBLISHED,
          authorId: '123',
          author: {
            displayName: 'jiko21',
          },
        },
      ],
    };
    const { container } = render(<PostList {...props} />);

    expect(container).toMatchSnapshot();
  });
});
