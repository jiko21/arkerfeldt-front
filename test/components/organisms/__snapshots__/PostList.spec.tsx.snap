// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList.spec.tsx should correctly render 1`] = `
.emotion-0 {
  list-style: none;
  padding: 0;
}

.emotion-0 li {
  margin-bottom: 6px;
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FCFCFC;
  border-radius: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 20px;
}

.emotion-2 input {
  display: none;
}

.emotion-2 input+label:before {
  border: 2px solid;
  border-color: #2d2d2d;
  box-sizing: border-box;
  content: '';
  display: block;
  height: 24px;
  width: 24px;
}

.emotion-2 input:checked+label:after {
  box-sizing: border-box;
  content: '';
  display: block;
  width: 18px;
  height: 9px;
  margin-top: -9px;
  top: 28px;
  left: 20px;
  border-bottom: 3px solid;
  border-left: 3px solid;
  border-color: #007AAE;
  position: absolute;
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.emotion-3 {
  margin-left: 16px;
}

.emotion-4 {
  color: #007AAE;
  cursor: pointer;
  font-size: 24px;
  font-weight: 600;
}

.emotion-5 {
  color: #2d2d2d;
  font-size: 14px;
}

.emotion-6 {
  color: #007AAE;
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
}

.emotion-7 {
  color: #2d2d2d;
  font-size: 14px;
  margin-left: 10px;
}

.emotion-8 {
  margin-top: auto;
  margin-left: auto;
}

.emotion-9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #C4C4C4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 24px;
  width: 80px;
}

.emotion-10 {
  color: #6A6A6A;
  font-size: 14px;
  font-weight: 700;
}

<div>
  <ul
    class="emotion-0"
  >
    <li>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          <input
            readonly=""
            type="checkbox"
          />
          <label
            data-testid="postItem-checkbox"
          />
        </div>
        <div
          class="emotion-3"
        >
          <a
            class="emotion-4"
          >
            〇〇は結婚していた？彼女の名前は××？
          </a>
          <div>
            <span
              class="emotion-5"
            >
              Created by
               
              <a
                class="emotion-6"
              >
                jiko21
              </a>
            </span>
            <span
              class="emotion-7"
            >
              Last modified 
              12/11/2020, 10:00:00 AM
            </span>
          </div>
        </div>
        <div
          class="emotion-8"
        >
          <div
            class="emotion-9"
          >
            <span
              class="emotion-10"
            >
              公開中
            </span>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          <input
            readonly=""
            type="checkbox"
          />
          <label
            data-testid="postItem-checkbox"
          />
        </div>
        <div
          class="emotion-3"
        >
          <a
            class="emotion-4"
          >
            〇〇は結婚していた？彼女の名前は××？
          </a>
          <div>
            <span
              class="emotion-5"
            >
              Created by
               
              <a
                class="emotion-6"
              >
                jiko21
              </a>
            </span>
            <span
              class="emotion-7"
            >
              Last modified 
              12/11/2020, 10:00:00 AM
            </span>
          </div>
        </div>
        <div
          class="emotion-8"
        >
          <div
            class="emotion-9"
          >
            <span
              class="emotion-10"
            >
              未公開
            </span>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
